src/executables/csql_support.c

```diff
@@ -249,84 +251,61 @@ csql_invoke_system (const char *command)
 int
 csql_invoke_system_editor (void)
 {
-  char *cmd = NULL;
-  char *fname = (char *) NULL;	/* pointer to temp file name */
-  FILE *fp = (FILE *) NULL;	/* pointer to stream */
-
   if (!iq_output_device_is_a_tty ())
     {
       csql_Error_code = CSQL_ERR_CANT_EDIT;
-      goto error;
-    }
-
-  /* create a temp file and open it */
-  fname = tempnam (NULL, NULL);
-
-  if (fname == NULL)
-    {
-      csql_Error_code = CSQL_ERR_OS_ERROR;
-      goto error;
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
     }
 
-  fp = fopen (fname, "w");
-  if (fp == NULL)
+  /* create an unique file in tmp folder and open it for writing */
+  auto[filename, fileptr] = filesys::open_temp_file ("csql_");
+  if (fileptr == NULL)
     {
       csql_Error_code = CSQL_ERR_OS_ERROR;
-      goto error;
+      nonscr_display_error (csql_Scratch_text, SCRATCH_TEXT_LEN);
+      return CSQL_FAILURE;
     }
+  filesys::auto_delete_file file_del (filename.c_str ());	//deletes file at scope end
+  filesys::auto_close_file file (fileptr);	//closes file at scope end (before the above file deleter); forget about fp from now on
```

Comment:
How is 'before the above file deleter' guaranteed? Is it using the nature of the stack to which local variables are allocated? If so, there is a dependency that must adhere to the order of declaration when using the two variables.